### PR TITLE
Added unit and runtime tests

### PR DESCRIPTION
**Pull request by DrDeano**
Also changed up panic call

### CODE REVIEW

File: src/kernel/arch/x86/irq.zig [hunk 1]
```diff
@@ -120,4 +159,144 @@ pub fn init() void {
...
+    // Set up
+    pic.initTest();
+    defer pic.freeTest();
+
+    pic.addTestParams("clearMask", u16(0));
+
+    // Pre testing
+    for (irq_handlers) |h| {
+        if (h) |_| {
+            std.debug.panic("Pre-testing failed");
```
Comment (SamTebbs33) [inline src/kernel/arch/x86/irq.zig]: This could be replaced with `expect(h == null)`

File: src/kernel/arch/x86/irq.zig [hunk 2]
```diff
@@ -120,4 +159,144 @@ pub fn init() void {
...
+    }
+
+    // Call function
+    try registerIrq(0, testFunction1);
+
+    // Post testing
+    for (irq_handlers) |h, i| {
+        if (h) |temp| {
+            if (i != 0) {
+                std.debug.panic("Post-testing failed");
```
Comment (SamTebbs33) [inline src/kernel/arch/x86/irq.zig]: Can this also be replaced with one of the `expect*` functions?

File: src/kernel/arch/x86/irq.zig [hunk 3]
```diff
@@ -120,4 +159,144 @@ pub fn init() void {
...
+
+    // Call function
+    try registerIrq(0, testFunction1);
+    expectError(IrqError.IrqExists, registerIrq(0, testFunction2));
+
+    // Post testing
+    for (irq_handlers) |h, i| {
+        if (h) |temp| {
+            if (i != 0) {
+                std.debug.panic("Post-testing failed");
```
Comment (SamTebbs33) [inline src/kernel/arch/x86/irq.zig]: See other comments

File: src/kernel/arch/x86/irq.zig [hunk 4]
```diff
@@ -120,4 +159,144 @@ pub fn init() void {
...
+    // Set up
+    pic.initTest();
+    defer pic.freeTest();
+
+    pic.addTestParams("clearMask", u16(0));
+
+    // Pre testing
+    for (irq_handlers) |h1| {
+        if (h1) |_| {
+            std.debug.panic("Pre-testing failed");
```
Comment (SamTebbs33) [inline src/kernel/arch/x86/irq.zig]: See other comments

File: src/kernel/arch/x86/irq.zig [hunk 5]
```diff
@@ -120,4 +159,144 @@ pub fn init() void {
...
+    openIrq(index, handler);
+}
+
+test "isValidIrq" {
+    comptime var i = 0;
+    inline while (i < NUMBER_OF_ENTRIES) : (i += 1) {
+        expectEqual(true, isValidIrq(i));
+    }
+
+    expectEqual(false, isValidIrq(200));
```
Comment (SamTebbs33) [inline src/kernel/arch/x86/irq.zig]: `expect(!isValidIrq(200))`?

File: src/kernel/arch/x86/irq.zig [hunk 6]
```diff
@@ -120,4 +159,144 @@ pub fn init() void {
...
+
+    idt.addTestParams("openInterruptGate", index, handler, ret);
+
+    openIrq(index, handler);
+}
+
+test "isValidIrq" {
+    comptime var i = 0;
+    inline while (i < NUMBER_OF_ENTRIES) : (i += 1) {
+        expectEqual(true, isValidIrq(i));
```
Comment (SamTebbs33) [inline src/kernel/arch/x86/irq.zig]: `expect(isValidIrq(i))`?

File: src/kernel/arch/x86/irq.zig (hunk 1 above)
Comment (DrDeano) [reply src/kernel/arch/x86/irq.zig]: Yes

**Pull request merged**